econml/ortho_forest.py

```diff
@@ -254,96 +321,54 @@ def const_marginal_effect(self, X):
         Theta : matrix , shape (n, d_t)
             Constant marginal CATE of each treatment for each sample.
         """
+        # TODO: Check performance
+        return np.asarray(self._predict(X))
+
+    def _predict(self, X, stderr=False):
         if not self.model_is_fitted:
             raise NotFittedError('This {0} instance is not fitted yet.'.format(self.__class__.__name__))
         X = check_array(X)
-        results = Parallel(n_jobs=self.n_jobs, verbose=3, backend='threading')(
-            delayed(self._pointwise_effect)(X_single) for X_single in X)
-        # TODO: Check performance
-        return np.asarray(results)
-
-    def _get_inference_options(self):
-        # Override the CATE inference options
-        # Add blb inference to parent's options
-        options = super()._get_inference_options()
-        options.update(blb=BLBInference)
-        options.update(auto=BLBInference)
-        return options
-
-    def _pointwise_effect(self, X_single, stderr=False):
-        """Calculate the effect for a one data point with features X_single.
-
-        Parameters
-        ----------
-        X_single : array-like, shape (d_x, )
-            Feature vector that captures heterogeneity for one sample.
-
-        stderr : boolean (default=False)
-            Whether to calculate the covariance matrix via bootstrap-of-little-bags.
-        """
+        results = Parallel(n_jobs=self.n_jobs, verbose=3)(
```

Comment:
Out of curiosity, have you compared performance vs. adding a `prefer="threads"` hint?  Given that most of the computation on each thread looks to be done via numpy, this might be faster since it could avoid copying large arrays across processes (but might be slower due to the GIL)

Reply:
it's slower. The point is that each process uses a sub-slice, which essentially is a new numpy array.

Reply:
My only left over is that processes have many times a big overhead and there is something related to the auto-batching that parallel does that slows down the first few periods. I will leave that to someone else to figure out if there is a better way to call parallel.